packages/core/src/controller/Controller.ts

```diff
@@ -244,13 +243,11 @@ export default class Controller {
     expiryStatus: ExpiryStatus;
     expiresAt: number;
   } => {
-    if (rest[0] === null) {
-      return nullResponse;
-    }
     const state = rest[rest.length - 1] as State<unknown>;
     const args = rest.slice(0, rest.length - 1) as Parameters<E['key']>;
-    const key = endpoint.key(...args);
-    const cacheResults = state.results[key];
+    const activeArgs = args?.[0] !== null;
```

Comment:
so this is basically testing for "the first item in `rest` is `null`"? what does "activeArgs" mean?

Reply:
open to naming suggestions - it distinguishes two argument mode:

[null] - meaning 'do nothing' (how to deal with no if/else in hooks)
...args - whatever args are defined by endpoint - so 'activate' the hook

Reply:
ok so `args.length === 1 && args[0] === null` indicates "do nothing"?
```suggestion
    const isActivation = args.length !== 1 || args[0] !== null;
```
(also, `rest.slice(0, -1)` on the line above)

Reply:
existing behavior has only checked first arg for null, so this would technically be a breaking change. What about isActive? activation doesn't seem like it describes how the hook works

Reply:
pushed change with isActive as name (and also updated its uses...since that's pretty important)